BankClient/Bank.cs

```diff
@@ -2,165 +2,204 @@
+            (listUsers.LastName == inputUser.LastName || inputUser.LastName == "") &&
+            (listUsers.PhoneNumber == inputUser.PhoneNumber || inputUser.PhoneNumber == "")).ToList();
+        return filterList;
     }
 
-    private List<UserInformation> SearchUsersInformation()
+    /// <summary>
+    /// Function saves filterList and call function for print
+    /// </summary>
+    /// <param name="filterList"></param>
+    /// <returns>groupOfUsers</returns>
+    public List<UserInformation> GetAllUsersInformation(List<UserInformation> filterList)
     {
-        Console.WriteLine("Example:Dmitry,Lapshin,0950095521");
-        Console.Write("Fiter: ");
-        var userfilter = Console.ReadLine().Split(',');
-        foreach (var param in userfilter)
+        var groupOfUsers = filterList;
+        
+        if (groupOfUsers != null)
         {
-            param.Trim();
+            
+            ShowListOfUsers(groupOfUsers);
         }
 
-        if (userfilter[0] == "" && userfilter.Length == 1)
+        return groupOfUsers;
+        
+    }
+
+    /// <summary>
+    /// Gets one user
+    /// </summary>
+    /// <param name="filterList">Takes <see cref="UserInformation"/> list which contains all user information</param>
+    /// <returns>Nothing</returns>
+
+    public UserInformation GetOneUserInformation(List<UserInformation> filterList)
+    {
+        if (filterList != null)
         {
-            ShowAllUsers();
+            var oneUserInformation = filterList[0];
+            ShowOneUser(oneUserInformation);
+            return oneUserInformation;
         }
-        var filterlist = Users.Where(x => x.FirstName == userfilter[0] || x.LastName == userfilter[0] || x.Phonenumber== userfilter[0] || x.ID.ToString() == userfilter[0]).ToList();
-        for (int i = 0; i < userfilter.Length; i++)
+        else
+        {
+            return null;
+        }
+
+    }
+    /// <summary>
+    /// Search user by ID
+    /// </summary>
+    /// <returns>Return user or null</returns>
+    public UserInformation GetUserByID()
+    {
+        string id = Console.ReadLine();
+        
+        foreach (var user in Users)
         {
-            filterlist = filterlist.Where(x => x.FirstName == userfilter[i] || x.LastName == userfilter[i] || x.Phonenumber == userfilter[i] ||  x.ID.ToString() == userfilter[0]).ToList();
+            if (user.ID.ToString() == id)
+            {
+                ShowOneUser(user);
+                return user;
+            }
         }
 
-        return filterlist;
+        return null;
     }
 
-    public void GetAllUsersInformation(List<UserInformation> filterlist)
+    /// <summary>
+    /// Change user information
+    /// </summary>
+    /// <param name="inputUsers">Takes <see cref="UserInformation"/> object which has all user information</param>
+    /// <returns>Nothing</returns>
+
+    public void ChangeUserInfomation(List<UserInformation> inputUsers)
```

Comment:
Let's rename it to `UpdateUserInfomation` and accept `UserInformation` as argument. We should update 1 user.

Reply:
Ok